db/etcd.go

```diff
@@ -569,259 +406,177 @@ func (etcddb etcdDB) InitTestRun(testUuid string, testAgentMap map[string]map[st
+			continue
+		}
 
+		testRunDataKey := fmt.Sprintf("%s/testdata", node.Key)
+		dataResp, err := etcddb.keysAPI.Get(context.Background(), testRunDataKey, nil)
+		if err != nil {
+			log.Errorf("Error retrieving testdata of agent in: %s", testUUID)
+			return nil, err
 		}
+		retMap[agentUUID] = dataResp.Node.Value
 	}
-	return ret_map
+
+	return retMap, nil
 }
 
 // WriteCleanTestData will write the post-test metrics data that has been cleaned up and
 // ready to be displayed or exported to the database
-func (etcddb etcdDB) WriteCleanTestData(testUuid string, testData string) {
+func (etcddb *etcdDB) WriteCleanTestData(testUUID string, testData string) error {
 
-	etcd_loc := fmt.Sprintf("http://%s:%s", etcddb.config.DB.IP, etcddb.config.DB.Port)
-	cfg := client.Config{
-		Endpoints: []string{etcd_loc},
-		Transport: client.DefaultTransport,
-		// set timeout per request to fail fast when the target endpoint is unavailable
-		HeaderTimeoutPerRequest: time.Second,
-	}
-	c, err := client.New(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-	kapi := client.NewKeysAPI(c)
+	log.Debugf("/todd/testruns/%s/cleandata/", testUUID)
 
-	log.Debugf("/todd/testruns/%s/cleandata/", testUuid)
+	keyStr := fmt.Sprintf("/todd/testruns/%s/cleandata/", testUUID)
 
-	key_str := fmt.Sprintf("/todd/testruns/%s/cleandata/", testUuid)
-
-	_, err = kapi.Set(
+	_, err := etcddb.keysAPI.Set(
 		context.Background(), // context
-		key_str,              // key
+		keyStr,               // key
 		testData,             // value
 		nil,                  //optional args
 	)
 	if err != nil {
 		log.Error("Problem setting object in etcd")
-		os.Exit(1)
+		return err
 	}
 
-	log.Infof("Wrote clean test data to test uuid: %s", testUuid)
+	log.Infof("Wrote clean test data to test uuid: %s", testUUID)
 
+	return nil
 }
 
 // GetCleanTestData will retrieve clean test data from the database
-func (etcddb etcdDB) GetCleanTestData(testUuid string) string {
-
-	etcd_loc := fmt.Sprintf("http://%s:%s", etcddb.config.DB.IP, etcddb.config.DB.Port)
-	cfg := client.Config{
-		Endpoints: []string{etcd_loc},
-		Transport: client.DefaultTransport,
-		// set timeout per request to fail fast when the target endpoint is unavailable
-		HeaderTimeoutPerRequest: time.Second,
-	}
-	c, err := client.New(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-	kapi := client.NewKeysAPI(c)
+func (etcddb *etcdDB) GetCleanTestData(testUUID string) (string, error) {
 
-	key_str := fmt.Sprintf("/todd/testruns/%s/cleandata", testUuid)
+	keyStr := fmt.Sprintf("/todd/testruns/%s/cleandata", testUUID)
 
-	log.Debug("Retrieving clean test data for ", testUuid)
+	log.Debug("Retrieving clean test data for ", testUUID)
 
-	resp, err := kapi.Get(context.Background(), key_str, &client.GetOptions{Recursive: true})
-	if err != nil {
-		log.Error(err)
-		log.Errorf("Error - empty test data: %s", testUuid)
-		return ""
+	resp, err := etcddb.keysAPI.Get(context.Background(), keyStr, &client.GetOptions{Recursive: true})
```

Comment:
I am good with including it in this patch as long as you provide some useful comments describing it's purpose in the follow-up.